docs/adr/0001-pipeline-logging-and-telemetry.md

```diff
@@ -0,0 +1,76 @@
+# Pipeline Logging and Telemetry
+
+## Status
+
+Proposed
+
+## Context
+
+Typically CI/CD pipelines use file-based logging or integrate with the logging features of the CI/CD platform to surface progress and diagnostic information.  This presents the following challenges:
+
+* Unstructured output
+* Portability issues across CI/CD platforms
+* Balancing the log verbosity in day-to-day usage, with the need for additional diagnostics in the event of an issue. This can result in having to re-run a failed pipeline in 'diagnostic' mode to get the required information, which makes it harder to troubleshoot inconsistently reproducible issues
+* Limited support for metrics & telemetry
+* Limited support for reporting & trend analysis over time
+
+It is proposed that the above issues are addressed by decoupling the logging activities from the CI/CD platform - such that the CI/CD platform becomes a consumer of the underlying logging system rather than the primary publisher of the log data.
+
+The following implementation options are considered below:
+
+* Azure Application Insights / Azure Log Analytics
+* Azure Data Lake Storage Gen2
+* Azure CosmosDB (serverless) 
+
+### Azure Application Insights / Azure Log Analytics
+This options sends all log messages to an AppInsights workspace.  It would require the types of log data produced by the CI/CD workload to be mapped to the application logging semantics offered by AppInsights.
+
+Pros:
+* Built-in semantics for tracking events, exceptions etc.
+* OOTB visualisation and query tools in the Azure Portal
+* .Net SDK reduces up-front integration effort
+
+Cons:
+* Price premium compared to purely data storage-based pricing of other options, both for ingestion >5GB/month and retention >90 days
+* Potential lag in data being available (potentially mitigated by Live Metrics Stream if supported for this scenario)
```

Comment:
Agree that the delay (circa 5 minutes) can be a frustration.  But would you be using this to debug live issues for example whilst getting a CI/CD new pipeline up and running for the first time?